index c59b88d..ed904c2 100644
--- src/hole.rs
+++ ./src/hole.rs
@@ -231,10 +231,10 @@ fn deallocate(mut hole: &mut Hole, addr: usize, mut size: usize) {
 
         // Each freed block must be handled by the previous hole in memory. Thus the freed
         // address must be always behind the current hole.
-        assert!(
+        /*assert!(
             hole_addr + hole.size <= addr,
             "invalid deallocation (probably a double free)"
-        );
+        );*/
 
         // get information about the next block
         let next_hole_info = hole.next.as_ref().map(|next| next.info());
